File: enode/src/main/java/org/enodeframework/domain/IAggregateRepositoryProxy.java
package org.enodeframework.domain;

import org.enodeframework.infrastructure.IObjectProxy;

import java.util.concurrent.CompletableFuture;

public interface IAggregateRepositoryProxy extends IObjectProxy {
    <T extends IAggregateRoot> CompletableFuture<T> getAsync(String aggregateRootId);
}

File: enode/src/main/java/org/enodeframework/domain/IAggregateRepositoryProxy.kt
package org.enodeframework.domain

import org.enodeframework.infrastructure.IObjectProxy
import java.util.concurrent.CompletableFuture

interface IAggregateRepositoryProxy : IObjectProxy {
    fun <T : IAggregateRoot?> getAsync(aggregateRootId: String): CompletableFuture<T>
}